doc/src/tutorials/physics/mechanics/duffing-example.rst

```diff
@@ -0,0 +1,148 @@
+- :math:`\dot{q}_1`: First time derivative of `q1`, representing the velocity of the Duffing oscillator
+- :math:`\dot{q}_2`: First time derivative of `q2`, representing the angular velocity of the pendulum
+- :math:`u_1`: Generalized speed associated with the Duffing oscillator
+- :math:`u_2`: Generalized speed associated with the pendulum
+- :math:`\dot{u}_1`: First time derivative of `u1`, representing the acceleration of the Duffing oscillator
+- :math:`\dot{u}_2`: First time derivative of `u2`, representing the angular acceleration of the pendulum
+
+Define Kinematics
+=================
+
+Define all the reference frames and points.
+
+   >>> N = me.ReferenceFrame('N')
+   >>> B = N.orientnew('B', 'axis', (q2, N.z))
+
+The angular velocity of the pendulum in the reference frame is:
+
+   >>> B.set_ang_vel(N, q2.diff() * N.z)
+   >>> B.ang_vel_in(N)
+   Derivative(q2(t), t)*N.z
+
+Locations and velocities of the Duffing Oscillator block and the pendulum are:
+
+   >>> O = me.Point('O')
+   >>> Block = O.locatenew('Block', q1 * N.y)
+   >>> Pendulum = Block.locatenew('Pendulum', l * B.y)
+
+O is a fixed point in the inertial reference frame.
+
+   >>> O.set_vel(N, 0)
+   >>> Block.set_vel(N, q1d * N.y)
+   >>> Pendulum.v2pt_theory(Block, N, B)
+   Derivative(q1(t), t)*N.y - l*Derivative(q2(t), t)*B.x
+
+   >>> me.init_vprinting()
+
+Define inertia and rigid bodies.
+Here, we assume a simple pendulum which consists of a bob of mass m hanging from a massless string of length l
+and fixed at a pivot point (Duffing Oscillator Block).
+
+   >>> I_block = me.inertia(N, M*(h**2 + d**2)/12, M*(w**2 + h**2)/12, M*(w**2 + d**2)/12)
+   >>> I_pendulum = me.inertia(B, 0, 0, 2*m*r**2/5)
+
+   >>> par_block = me.RigidBody('block', Block, N, M, (I_block, Block))
+   >>> par_pendulum = me.RigidBody('pendulum', Pendulum, B, m, (I_pendulum, Pendulum))
+
+Define Forces Using a Custom Actuator
+=====================================
+
+We define a custom actuator class, `DuffingDamperPathway`, to calculate the forces acting on the system.
+This class combines the forces from a Duffing spring and a damper to model the dynamics of the Duffing oscillator block.
+We also consider gravity as a force acting on the pendulum.
+
+   >>> class DufPenPathway(me.ActuatorBase):
+   ...     def __init__(self, P1, P2, P3, linear_stiffness, nonlinear_stiffness, damping, pendulum_mass, block_mass, g):
+   ...         self.P1 = P1
+   ...         self.P2 = P2
+   ...         self.P3 = P3
+   ...         self.k1 = linear_stiffness
+   ...         self.k2 = nonlinear_stiffness
+   ...         self.c1 = damping
+   ...         self.m = pendulum_mass
+   ...         self.M = block_mass
+   ...         self.g = g
+   ...
+   ...         self.pathway1 = me.LinearPathway(self.P1, self.P2)
+   ...         self.duffing_spring = me.DuffingSpring(self.k1, self.k2, self.pathway1, 0)
+   ...         self.pathway2 = me.LinearPathway(self.P2, self.P3)
```

Comment:
Why would the pathways be different for the spring and the damper? Aren't they in parallel?